applicationconfig/client/src/test/java/com/azure/applicationconfig/ConfigurationClientTest.java

```diff
@@ -60,36 +60,32 @@
     @Before
     public void beforeTest() throws Exception {
         final TestMode testMode = getTestMode();
-        final String playbackUri = getPlaybackUri(testMode);
-        final HttpPipelinePolicy loggingPolicy = new HttpLoggingPolicy(HttpLogDetailLevel.BODY_AND_HEADERS);
-        final String connectionString;
-        final HttpPipeline pipeline;
 
         interceptorManager = InterceptorManager.create(testName.getMethodName(), testMode);
 
         if (interceptorManager.isPlaybackMode()) {
             logger.info("PLAYBACK MODE");
 
-            connectionString = "endpoint=" + playbackUri + ";Id=0000000000000;Secret=MDAwMDAw";
+            final String playbackUri = getPlaybackUri(testMode);
+            final String connectionString = "endpoint=" + playbackUri + ";Id=0000000000000;Secret=MDAwMDAw";
 
-            List<HttpPipelinePolicy> policies = ConfigurationClient.getDefaultPolicies(connectionString);
-            policies.add(loggingPolicy);
-
-            pipeline = new HttpPipeline(interceptorManager.getPlaybackClient(), policies);
+            client = ConfigurationClient.builder(new ConfigurationClientCredentials(connectionString))
+                    .withHttpClient(interceptorManager.getPlaybackClient())
+                    .withHttpLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS)
+                    .build();
         } else {
             logger.info("RECORD MODE");
 
-            connectionString = System.getenv("AZCONFIG_CONNECTION_STRING");
+            final String connectionString = System.getenv("AZCONFIG_CONNECTION_STRING");
             Objects.requireNonNull(connectionString, "AZCONFIG_CONNECTION_STRING expected to be set.");
 
-            List<HttpPipelinePolicy> policies = ConfigurationClient.getDefaultPolicies(connectionString);
-            policies.add(interceptorManager.getRecordPolicy());
-            policies.add(loggingPolicy);
-
-            pipeline = new HttpPipeline(HttpClient.createDefault().wiretap(true), policies);
+            client = ConfigurationClient.builder(new ConfigurationClientCredentials(connectionString))
+                    .withHttpClient(HttpClient.createDefault().wiretap(true))
+                    .withHttpLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS)
+                    .withPolicy(interceptorManager.getRecordPolicy())
```

Comment:
We should discuss the 'with' naming pattern. I've observed within Microsoft that two patterns appear to exist for fluent API, `withPolicy(..)` and `policy(..)`.

Reply:
Is there one that is more predominantly used? I went with the naming that I saw in a random class here: https://github.com/Azure/azure-libraries-for-java/

Reply:
So funnily enough they are pretty much the 'land of the with' :-) I would say that from what I've seen closer to our home in azure-sdk-for-java we typically don't use 'with'. I would like to get any input from @anuchandy and @jianghaolu though, because theoretically they straddle both sides now and I'm not clear why this discrepancy arose.

If it were me, well, I come from the land where everything should be a standard, non-fluent setter method :-) I would probably say without 'with' is right for us, as that is what we see in azure-common today.

Reply:
But please disagree with me. I hold almost no opinion here and want what is right.

Reply:
I've done more research and I now feel more opinionated :-) I don't like the 'with' naming pattern.

Reply:
I don't have an opinion. What was the rationale you had for having just foo() vs withFoo()? I see a few other large SDKs use setFoo() too.

Reply:
I've updated the code. It does look a lot cleaner. :)

Reply:
same here, don't have a strong opinion on `foo(arg)` vs `withFoo(arg)`. I too liked `foo(arg)` so followed this in client-common.